README.md

```diff
@@ -11,7 +11,27 @@ I have not provided any test code, but you can base yours on test code from
 other exercises. Your tests must check the correctness of the result of running
 the function and run automatically when you commit through a GitHub action.
 
-## Runtime Analysis
+# Runtime Analysis
 
 What is the worst-case big $\Theta$ complexity of your implementation? Add your
 answer, including your reasoning, to this markdown file.
+
+
+## My Runtime Analysis
+
+When my hasCycle() algorithm runs, it first checks if the length of the graph is 0 or 1, which if it is, the algorithm runs in constant time, because graphs of those lengths don't have any cycles.
+
+If the length is greater than 1, it reaches a for loop which cycles through every node within the graph and checks all possible pathways from that node. This for loop will execute |V| times since it will only iterate through all the nodes, and not every possible edge within the graph.
+
+We then reach a recursive call which essentially executes a depth first search. Within this recursion is another for loop which will iterate through all the paths of the given current node. In the worst case scenerio, this will take |V|^2 to execute, since I am using an adjacency matrix representation and every possible edge is checked.
```

Comment:
Not sure if I'm following here -- you're only going to consider each node once, right? Otherwise you would have duplicates in `foundNodes`.